spyder/api/plugins/new_api.py

```diff
@@ -572,204 +896,310 @@ def set_conf(
+            If ``None``, the default, then the value of :attr:`CONF_SECTION`
+            is used.
+
+        Returns
+        -------
+        None
+
+        Raises
+        ------
+        SpyderAPIError
+            If ``section`` is not passed and :attr:`CONF_SECTION` is not
+            defined.
         """
         if self._conf is not None:
             section = self.CONF_SECTION if section is None else section
             if section is None:
                 raise SpyderAPIError(
-                    'A spyder plugin must define a `CONF_SECTION` class '
-                    'attribute!'
+                    "A spyder plugin must define a `CONF_SECTION` class "
+                    "attribute!"
                 )
             self._conf.restore_notifications(section, option)
 
     @Slot(str)
     @Slot(str, int)
-    def show_status_message(self, message, timeout=0):
+    def show_status_message(self, message: str, timeout: int = 0) -> None:
         """
-        Show message in status bar.
+        Show a message in the Spyder status bar.
 
         Parameters
         ----------
         message: str
-            Message to display in the status bar.
+            The message to display in the status bar.
         timeout: int
-            Amount of time to display the message.
+            The amount of time, in milliseconds, to display the message.
+            If ``0``, the default, the message will be shown until a plugin
+            calls :meth:`!show_status_message` again.
+
+        Returns
+        -------
+        None
         """
         self.sig_status_message_requested.emit(message, timeout)
 
-    def before_long_process(self, message):
+    def before_long_process(self, message: str) -> None:
         """
-        Show a message in main window's status bar and change the mouse
-        pointer to Qt.WaitCursor when starting a long process.
+        Perform actions required before starting a longer-running process.
+
+        Shows a message in the main window's status bar,
+        changes the mouse pointer to a ``WaitCursor``, and starts a spinner.
 
         Parameters
         ----------
         message: str
             Message to show in the status bar when the long process starts.
+
+        Returns
+        -------
+        None
         """
         if message:
             self.show_status_message(message)
 
         QApplication.setOverrideCursor(QCursor(Qt.WaitCursor))
         QApplication.processEvents()
 
-    def after_long_process(self, message=""):
+    def after_long_process(self, message: str = "") -> None:
         """
-        Clear main window's status bar after a long process and restore
-        mouse pointer to the OS deault.
+        Perform actions required after starting a longer-running process.
+
+        Clears the message in the Spyder main window's status bar,
+        restores the mouse pointer to the OS default, and stops spinner.
```

Comment:
```suggestion
        Clears the message in the Spyder main window's status bar
        and restores the mouse pointer to the OS default.
```

This is not correct (and it's not mentioned in the current docstring).

Reply:
Yeah I didn't see it in the proximate code here, it just didn't make sense to me why it wouldn't parallel its `before_` variant.

In fact, I don't see any different code in the `before_` callback that would start the spinner, so should it not be mentioned there either? Or what's the deal here?

Reply:
Yup, turns out it should be removed from the above too.